src/lib/tuning/greedy_selector.cpp

```diff
@@ -0,0 +1,158 @@
+#include "greedy_selector.hpp"
+
+#include <algorithm>
+#include <limits>
+#include <list>
+
+#include "utils/logging.hpp"
+
+namespace opossum {
+
+/*
+ * This subroutine scans the least desirable available choices until the
+ * accumulated reject_cost() is lower than a required value.
+ * If the thereby accumulated reject_desirability is not lower than a threshold
+ * an iterator to the choice one past the subsequence to reject is returned.
+ * Otherwise the end() iterator is returned, that the constraints are not satisfied
+ * by any continuous subsequence starting at sorted_choices.begin().
+ */
+const std::list<std::shared_ptr<TuningChoice>>::const_iterator sacrifice_choices(
```

Comment:
could be in an anonymous namespace as its private to this cpp. Also returning const values doesn't make much sense